src/runtime/hexagon_remote/halide_remote.cpp

```diff
@@ -326,12 +354,21 @@ int halide_hexagon_remote_power_hvx_off() {
     return 0;
 }
 
-int halide_hexagon_remote_get_symbol_v2(handle_t module_ptr, const char* name, int nameLen,
-                                        handle_t *sym_ptr) {
-    *sym_ptr = reinterpret_cast<handle_t>(obj_dlsym(reinterpret_cast<elf_t*>(module_ptr), name));
+int halide_hexagon_remote_get_symbol_v2(handle_t module_ptr, const char* name, int nameLen, int use_shared_object, handle_t *sym_ptr) {
+    if (use_shared_object) {
+       *sym_ptr = reinterpret_cast<handle_t>(dlsym(reinterpret_cast<elf_t*>(module_ptr), name));
+    } else {
+        *sym_ptr= reinterpret_cast<handle_t>(obj_dlsym(reinterpret_cast<elf_t*>(module_ptr), name));
+    }
     return *sym_ptr != 0 ? 0 : -1;
 }
 
+handle_t halide_hexagon_remote_get_symbol_v3(handle_t module_ptr, const char* name, int nameLen, bool use_shared_object) {
```

Comment:
I don't this this function is necessary. The IDL only exposes one of these get_symbol functions, so unless it is called from somewhere inside the runtime, it's dead code and should be removed.